tools/RestfulApiWebService/OperationOnDB/Controllers/ApiFileController.cs

```diff
@@ -0,0 +1,75 @@
+﻿using DB.Dao;
+using DB.Models;
+using DB.Utility;
+using Microsoft.DocAsCode.Build.Engine;
+using Microsoft.DocAsCode.Common;
+using Newtonsoft.Json;
+using OperationOnDB.Filters;
+using System.IO;
+using System.Web;
+using System.Web.Hosting;
+using System.Web.Http;
+
+namespace OperationOnDB.Controllers
+{
+    [ApiBasicAuthenticationAttribute("user", "user",
+    BasicRealm = "localhost")]
+    [RoutePrefix("uids")]
+    public class ApiFileController : ApiController
+    {
+        private XrefSpecDBContext _db = new XrefSpecDBContext();
+
+
+        [HttpPost]
+        [Route("add")]
+        public string UploadFiles()
+        {
+            int uploadedFileNumber = 0;
+            string fileBasePath = HostingEnvironment.MapPath("~/uploads");
+
+            HttpFileCollection hfc = HttpContext.Current.Request.Files;
+
+            for(int count = 0; count < hfc.Count; count++)
+            {
+                HttpPostedFile hpf = hfc[count];
+                if(hpf.ContentLength > 0)
+                {
+                    string FullPath = Path.Combine(fileBasePath, Path.GetFileName(hpf.FileName));
+                    if (!File.Exists(FullPath))
+                    {
+                        hpf.SaveAs(FullPath);
```

Comment:
no need to save, deserialize from `hpf.InputStream` directly. When changing to aspnet core, use `IFormFile` and read from `IFromFile.OpenReadStream` instead, referencing https://www.mikesdotnetting.com/article/288/uploading-files-with-asp-net-core-1-0-mvc